File: domain/src/main/java/com/duyp/architecture/clean/android/powergit/domain/utils/CommonUtil.java
package com.duyp.architecture.clean.android.powergit.domain.utils;

public final class CommonUtil {

    /**
     * check if given input is either null or zero length.
     *
     * @param str input
     *
     * @return true if input is empty, otherwise false.
     */
    public static boolean isEmpty(final CharSequence str) {
        return str == null || str.length() == 0;
    }

    /**
     * Check if at least 1 element of given arrays is empty
     *
     * @return true if at least 1 element is empty
     */
    public static boolean isEmpty(final CharSequence... str) {
        for(CharSequence s : str) {
            if (isEmpty(s)) {
                return true;
            }
        }
        return false;
    }

    public static boolean isNumeric(final String str) {
        return str.matches("[-+]?\\d*\\.?\\d+");
    }
}

File: domain/src/main/java/com/duyp/architecture/clean/android/powergit/domain/utils/CommonUtil.kt
package com.duyp.architecture.clean.android.powergit.domain.utils

class CommonUtil {

    companion object {
        /**
         * check if given input is either null or zero length.
         *
         * @param str input
         *
         * @return true if input is empty, otherwise false.
         */
        fun isEmpty(str: String?): Boolean {
            return str == null || str.isEmpty()
        }

        /**
         * Check if at least 1 element of given arrays is empty
         *
         * @return true if at least 1 element is empty
         */
        fun isEmpty(vararg str: String?): Boolean {
            for (s in str) {
                if (s == null || s.isEmpty()) {
                    return true
                }
            }
            return false
        }

        fun isNumeric(str: String): Boolean {
            return str.matches("[-+]?\\d*\\.?\\d+".toRegex())
        }
    }

}
